pkg/plugins/instrumentationutils/request_status.go

```diff
@@ -0,0 +1,104 @@
+package instrumentationutils
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+)
+
+type RequestStatus int
+
+const (
+	RequestStatusOK RequestStatus = iota
+	RequestStatusCancelled
+	RequestStatusError
+)
+
+func (status RequestStatus) String() string {
+	names := [...]string{"ok", "cancelled", "error"}
+	if status < RequestStatusOK || status > RequestStatusError {
+		return ""
+	}
+
+	return names[status]
+}
+
+func RequestStatusFromError(err error) RequestStatus {
+	status := RequestStatusOK
+	if err != nil {
+		status = RequestStatusError
+		if errors.Is(err, context.Canceled) || grpcstatus.Code(err) == grpccodes.Canceled {
+			status = RequestStatusCancelled
+		}
+	}
+
+	return status
+}
+
+func RequestStatusFromErrorString(errString string) RequestStatus {
+	status := RequestStatusOK
+	if errString != "" {
+		status = RequestStatusError
+		if strings.Contains(errString, context.Canceled.Error()) || strings.Contains(errString, "code = Canceled") {
```

Comment:
Bit of a hack since we don't have access to the actual error type, just a string.